src/System.ComponentModel.Annotations/src/System/ComponentModel/DataAnnotations/AssociatedMetadataTypeTypeDescriptionProvider.cs

```diff
@@ -0,0 +1,32 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+namespace System.ComponentModel.DataAnnotations
+{
+    public class AssociatedMetadataTypeTypeDescriptionProvider : TypeDescriptionProvider
+    {
+        private Type _associatedMetadataType;
+        public AssociatedMetadataTypeTypeDescriptionProvider(Type type)
+            : base(TypeDescriptor.GetProvider(type))
+        {
+        }
+
+        public AssociatedMetadataTypeTypeDescriptionProvider(Type type, Type associatedMetadataType)
+            : this(type)
+        {
+            if (associatedMetadataType == null)
+            {
+                throw new ArgumentNullException("associatedMetadataType");
```

Comment:
Use nameof?

Reply:
I really appreciate that. I'm gonna provide a list of changes which are applicable here. After you approved them, I'll apply them too.